### PR TITLE
refactor: add WatcherInterface helper struct

### PR DESCRIPTION
**Pull request by bartlomieju**
Prerequisite for https://github.com/denoland/deno/pull/20876

### CODE REVIEW

File: cli/util/file_watcher.rs [hunk 1]
```diff
@@ -109,26 +108,85 @@ fn create_print_after_restart_fn(clear_screen: bool) -> impl Fn() {
   }
 }
 
+// TODO(bartlomieju): this is a poor name; change it
+/// And interface to interact with Deno's CLI file watcher.
+pub struct WatcherInterface {
```
Comment (bartlomieju) [inline cli/util/file_watcher.rs:113]: @dsherret any suggestions for the name of this struct?

File: cli/util/file_watcher.rs [hunk 2]
```diff
@@ -109,26 +109,87 @@ fn create_print_after_restart_fn(clear_screen: bool) -> impl Fn() {
...
 }
 
+// TODO(bartlomieju): this is a poor name; change it
+/// An interface to interact with Deno's CLI file watcher.
+pub struct WatcherInterface {
+  /// Send a list of paths that should be watched for changes.
+  pub paths_to_watch_tx: tokio::sync::mpsc::UnboundedSender<Vec<PathBuf>>,
+
+  /// Listen for a list of paths that were changed.
+  pub changed_paths_rx: tokio::sync::broadcast::Receiver<Option<Vec<PathBuf>>>,
```
Comment (nayeemrmn) [inline cli/util/file_watcher.rs:119]: Is this used anywhere?

File: cli/util/file_watcher.rs (hunk 1 above)
Comment (deckchairlabs) [reply cli/util/file_watcher.rs:114]: ChangeNotifierInterface ?

File: cli/util/file_watcher.rs (hunk 2 above)
Comment (bartlomieju) [reply cli/util/file_watcher.rs:119]: Not yet, will be used by #20876

File: cli/util/file_watcher.rs (hunk 1 above)
Comment (nayeemrmn) [reply cli/util/file_watcher.rs:114]: Maybe like `WatcherChannel`. It's bidirectional communication between inside and outside the watched operation

**Review by dsherret — APPROVED** [review]
LGTM, but I think us exposing all these channels in higher level code is not ideal. I believe we could make the low level channels not exposed to other parts of the code.

File: cli/util/file_watcher.rs [hunk 3]
```diff
@@ -109,26 +109,87 @@ fn create_print_after_restart_fn(clear_screen: bool) -> impl Fn() {
...
+
+  /// Send a message to force a restart.
+  pub restart_tx: tokio::sync::mpsc::UnboundedSender<()>,
```
[the remark this answers is not in the document]
Comment (bartlomieju) [reply cli/util/file_watcher.rs]: Done

**Pull request merged**